boards/shared/Test/Src/Test_ErrorTable.cpp

```diff
@@ -754,3 +758,450 @@ TEST_F(
         ASSERT_TRUE(App_SharedError_IsBoardInList(&board_list, board));
     }
 }
+
+TEST_F(SharedErrorTableTest, process_bms_non_critical_errors)
+{
+    // A list of BMS non-critical error IDs. This must be maintained manually as
+    // errors are added to/removed from the DBC!
+    std::vector<enum ErrorId> bms_non_critical_error_ids = {
+        BMS_NON_CRITICAL_STACK_WATERMARK_ABOVE_THRESHOLD_TASK1HZ,
+        BMS_NON_CRITICAL_STACK_WATERMARK_ABOVE_THRESHOLD_TASK1KHZ,
+        BMS_NON_CRITICAL_STACK_WATERMARK_ABOVE_THRESHOLD_TASKCANRX,
+        BMS_NON_CRITICAL_STACK_WATERMARK_ABOVE_THRESHOLD_TASKCANTX,
+        BMS_NON_CRITICAL_WATCHDOG_TIMEOUT
+    };
+
+    // Each struct member that has a non-zero value indicates an error that is
+    // set. For this test, we want every error to be set. Instead of setting
+    // each struct member manually, we're going to "cheat" using memset().
+    struct CanMsgs_bms_non_critical_errors_t data;
```

Comment:
Initially, I wrote assigning each struct member manually but that was incredibly cumbersome and error-prone (For example, we might forget to assign a struct member). I think using `memset` here is justified, especially with the explanation in the comments.